### PR TITLE
[Merged by Bors] - feat: add div/mod/dvd on Nat/Int to `norm_num`

### PR DESCRIPTION
**Pull request by digama0**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/NormNum/Basic.lean [hunk 1]
```diff
@@ -543,3 +543,28 @@ such that `norm_num` successfully recognises both `a` and `b`. -/
...
+theorem isNat_dvd_false : {a b : ℕ} → {a' b' c : ℕ} →
+    IsNat a a' → IsNat b b' → Nat.mod b' a' = Nat.succ c → ¬a ∣ b
+  | _, _, _, _, c, ⟨rfl⟩, ⟨rfl⟩, e => mt Nat.mod_eq_zero_of_dvd (e.symm ▸ Nat.succ_ne_zero c :)
+
+/-- The `norm_num` extension which identifies expressions of the form `(a : ℕ) | b`,
+such that `norm_num` successfully recognises both `a` and `b`. -/
+@[norm_num (_ : ℕ) ∣ _] def evalNatDvd : NormNumExt where eval {u α} e := do
+  let .app (.app f (a : Q(ℕ))) (b : Q(ℕ)) ← whnfR e | failure
+  -- We trust that the default instance for `Dvd` is `Nat.dvd` when the first parameter is `ℕ`.
+  guard <|← withNewMCtxDepth <| isDefEq f q(Dvd.dvd (α := ℕ))
```
Comment (eric-wieser) [inline Mathlib/Tactic/NormNum/Basic.lean:560]: Doesn't "we trust" here mean "we assert that"?

File: Mathlib/Tactic/NormNum/Basic.lean [hunk 2]
```diff
@@ -543,3 +543,28 @@ such that `norm_num` successfully recognises both `a` and `b`. -/
   have nc : Q(ℕ) := mkRawNatLit (na.natLit! / nb.natLit!)
   haveI' : Nat.div $na $nb =Q $nc := ⟨⟩
   return .isNat sℕ nc q(isNat_natDiv $pa $pb (.refl $nc))
+
+theorem isNat_dvd_true : {a b : ℕ} → {a' b' : ℕ} →
+    IsNat a a' → IsNat b b' → Nat.mod b' a' = nat_lit 0 → a ∣ b
```
Comment (eric-wieser) [inline Mathlib/Tactic/NormNum/Basic.lean:548]: What's the heuristic for using `nat_lit` vs `OfNat.ofNat`?

**Comment by kmill** [comment]
Here are some additional tests I'd written before I noticed your PR. The numbers don't need to be quite so big, but I copied them from a GCD test.
```
example : 2 ∣ 4 := by norm_num1
example : ¬ 2 ∣ 5 := by norm_num1
example : 553105253 ∣ 553105253 * 776531401 := by norm_num1
example : ¬ 553105253 ∣ 553105253 * 776531401 + 1 := by norm_num1

example : (2 : ℤ) ∣ 4 := by norm_num1
example : ¬ (2 : ℤ) ∣ 5 := by norm_num1
example : (553105253 : ℤ) ∣ 553105253 * 776531401 := by norm_num1
example : ¬ (553105253 : ℤ) ∣ 553105253 * 776531401 + 1 := by norm_num1
```

File: Mathlib/Tactic/NormNum/Basic.lean (hunk 1 above)
Comment (digama0) [reply Mathlib/Tactic/NormNum/Basic.lean:562]: yes, this comment is copy-pasted and seems wrong

File: Mathlib/Tactic/NormNum/Basic.lean (hunk 2 above)
Comment (digama0) [reply Mathlib/Tactic/NormNum/Basic.lean:550]: Using simpler terms means less work for the kernel, which is going to be normalizing the LHS to a nat literal.

**Comment by semorrison** [comment]
bors merge

File: .vscode/settings.json [hunk 3]
```diff
@@ -7,5 +7,8 @@
   "files.insertFinalNewline": true,
   "files.trimFinalNewlines": true,
   "files.trimTrailingWhitespace": true,
-  "githubPullRequests.ignoredPullRequestBranches": ["master"]
+  "githubPullRequests.ignoredPullRequestBranches": [
+    "master"
+  ],
+  "git.ignoreLimitWarning": true
```
Comment (digama0) [inline .vscode/settings.json:13]: @semorrison Is this intentional?

Comment (semorrison) [reply .vscode/settings.json:13]: No, it wasn't, I didn't realise I did that (nor did I make the edit manually...)

**Pull request closed without merging**